aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -178,6 +180,31 @@ module aptos_framework::block {
         );
     }
 
+    ///  Emit a `NewBlockEvent` event. This function will be invoked by write set script directly to generate the
+    ///  new block event for WriteSetPayload.
+    public fun emit_writeset_block_event(framework_signer: &signer) acquires BlockResource {
+        system_addresses::assert_aptos_framework(framework_signer);
+
+        let block_metadata_ref = borrow_global_mut<BlockResource>(@aptos_framework);
+        block_metadata_ref.height = event::counter(&block_metadata_ref.new_block_events);
+
+        let genesis_id = @0x0;
+        emit_new_block_event(
+            framework_signer,
+            &mut block_metadata_ref.new_block_events,
+            NewBlockEvent {
+                hash: genesis_id,
+                epoch: reconfiguration::current_epoch(),
+                round: MAX_U64,
+                height: block_metadata_ref.height,
+                previous_block_votes_bitvec: vector::empty(),
+                proposer: @vm_reserved,
+                failed_proposer_indices: vector::empty(),
+                time_microseconds: 0,
```

Comment:
Is this a valid value to use for time?

Reply:
And here as well

Reply:
We can use the current time here I think.

Reply:
The current time is set by block metadata tx. Since this is used in the context of a writeset where there's no block metadata tx, the timestamp would be the same as the previous block's timestamp. That still doesn't sound right but I'll let @zekun000 double check here as I don't have enough context on the discussion on writeset tooling design

Reply:
we probably should use current timestamp instead of 0, we were discussing to use real timestamp for genesis too